packages/main/src/components/ObjectPage/index.tsx

```diff
@@ -376,86 +377,150 @@ const ObjectPage: FC<ObjectPagePropTypes> = forwardRef((props: ObjectPagePropTyp
+
+        const threshold = 64;
+        const baseScrollValue =
+          activeContainer.current === contentContainer.current
+            ? e.target.scrollTop
+            : getProportionateScrollTop(activeInnerContainer, passiveInnerContainer, e.target.scrollTop);
+
+        const shouldBeCollapsed = baseScrollValue > threshold;
         if (collapsedHeader !== shouldBeCollapsed) {
-          // contentContainer.current.removeEventListener('scroll', onScroll);
+          lastScrolledContainer.current = activeContainer.current;
           if (shouldBeCollapsed) {
-            expandedHeaderHeight.current = innerHeaderHeight - 45;
-            collapsedHeaderFiller.current.style.height = `${expandedHeaderHeight.current}px`;
+            collapsedHeaderFiller.current.style.height = `${64}px`;
           } else {
             collapsedHeaderFiller.current.style.height = `${0}px`;
           }
+          lastScrolledContainer.current = activeContainer.current;
+          removeScrollEvent(contentContainer, stableContentOnScrollRef);
+          removeScrollEvent(scrollBar, stableBarOnScrollRef);
           setCollapsedHeader(shouldBeCollapsed);
         } else {
-          scrollBar.current.scrollTop = collapsedHeader
-            ? e.target.scrollTop
-            : getProportionateScrollTop(e.target.scrollTop);
+          const newScrollValue =
+            collapsedHeader && e.target.scrollTop > threshold + 50
+              ? e.target.scrollTop
+              : getProportionateScrollTop(activeInnerContainer, passiveInnerContainer, e.target.scrollTop);
+
+          passiveContainer.current.scrollTop = newScrollValue;
           scroller.current.scroll(e);
         }
+      }
+    },
+    [
+      innerHeader.current,
+      collapsedHeader,
+      contentContainer.current,
+      collapsedHeaderFiller.current,
+      setCollapsedHeader,
+      scrollBar.current,
+      scroller.current
+    ]
+  );
+
+  useEffect(() => {
+    if (!isMounted) return;
+    adjustDummyDivHeight().then(() => {
+      if (!hideHeaderButtonPressed.current) {
+        removeScrollEvent(contentContainer, stableContentOnScrollRef);
+        removeScrollEvent(scrollBar, stableBarOnScrollRef);
+        if (lastScrolledContainer.current === contentContainer.current) {
+          contentContainer.current.scrollTop = collapsedHeader ? 64 + 2 : 64 - 2;
+        } else {
+          contentContainer.current.scrollTop = collapsedHeader ? 64 + 2 : 64 - 2;
+          scrollBar.current.scrollTop = getProportionateScrollTop(
+            contentScrollContainer,
+            innerScrollBar,
+            contentContainer.current.scrollTop
+          );
+        }
+        requestAnimationFrame(() => {
+          bindScrollEvent(contentContainer, stableContentOnScrollRef);
+          bindScrollEvent(scrollBar, stableBarOnScrollRef);
+        });
+      }
+      hideHeaderButtonPressed.current = false;
+    });
+  }, [collapsedHeader]);
+
+  useMemo(() => {
```

Comment:
wouldn't `useEffect` be the better fit here? Or do you need it before rendering?

Reply:
should be an effect, you´re right.